Student Sketches and Answers.jl

```diff
@@ -27,238 +26,320 @@ begin
-	    		linewidth = 5,
-	    		background_color = :black,
-			    markershape = :circle,
-	    		markercolor = :deepskyblue,
-	    		markersize = 5,
-	    		markerstrokecolor = :skyblue,
-	 			legend=false)
-		
-	    title!("Timestamp: $timestamp")
+	
+	title!("Timestamp: $(min(end_stamp,begin_stamp+Dates.Millisecond(t)))")
+end
 
-		# {{{ Having p here doesn't do anything I don't think}}}
-	    p  # Return the plot for the current frame
-	end
+# ╔═╡ 86314c50-e636-4ee9-85f8-d52106ff4fdf
+format_dt(timestamp) = DateTime(replace(timestamp, "+00:00" => ""), DateFormat("yyyy-mm-dd HH:MM:SS.ssss"))
 
-		
-	# Display the animation  {{{ I think just return the anim and then call gif outside the function }}}
-	return gif(anim),p,sy
-	end
+# ╔═╡ cb2f8407-c2a5-4b51-8036-5ea9094b467f
+function format_timestamp_data(timestamp_data)
+    formatted_timestamp_data = []
+    for timestamp in timestamp_data
+		timestamp=timestamp[1:23]
+		if timestamp[20]=='+'
+			timestamp=timestamp[1:19]*".000"
+		end
+		formatted_timestamp=format_dt(timestamp)
+        push!(formatted_timestamp_data, format_dt(timestamp))
+    end
+    return formatted_timestamp_data
 end
 
-# ╔═╡ 693de731-e00b-4e99-b74a-33583f0f05b8
+# ╔═╡ d0c7bd76-ae05-48ad-a76b-ecf2737143da
 begin
-			sorted_data = filter(row -> row.uid == selected_student, DF)
-		end_timestamp = student_row[student_row.q_id .== 110, :].timestamp[1]
-		animation_data = filter(row -> row.timestamp <= end_timestamp, sorted_data)
-	anim,p,sy=PLOT(animation_data)
-	anim
+	a_df = CSV.read("student_answers_W2023.csv", DataFrame) #answer dataframe
+	s_df = CSV.read("IAP_sketch_data.csv", DataFrame) #sketch dataframe
+	a_df.timestamp = strip.(a_df.timestamp)
+	sort!(s_df, [:uid, :timestamp])
+	a_df = filter(row -> row.timestamp >= "2023-01-01 00:00:00.000000+00:00", a_df)
+	s_df = filter(row -> row.timestamp >= "2023-01-01 00:00:00.000000+00:00", s_df)
+	s_df.timestamp = format_timestamp_data(s_df.timestamp)	
+	a_df.timestamp = format_timestamp_data(a_df.timestamp)	
 end
 
-# ╔═╡ 42e1166c-5662-4950-94a1-05d61777e61b
-p
-
-# ╔═╡ d1fb2a00-d260-4814-90f0-f968b590618c
-Markdown.parse("""The y vector at the end of answer-1 is 
+# ╔═╡ 0584d086-883d-49bf-b1e3-b01e0ba33a12
+@bind selected_student Select(unique(a_df.uid))
 
-$sy """)
+# ╔═╡ bb252b8f-aa11-4567-8f09-fd6c9e6f76ff
+begin	
+	student_answers = a_df[a_df.uid .== selected_student, :]    #filtered answer data frame containing the entries of the selected student
+	sketch_url_110 = student_answers[student_answers.q_id .== 110, :].sketch_url[1]; #url for Q1
+	sketch_url_111 = student_answers[student_answers.q_id .== 111, :].sketch_url[1]; #url for Q2
+	text_answer_110 = student_answers[student_answers.q_id .== 110, :].text_answer[1]; #ans for Q1
+	text_answer_111 = student_answers[student_answers.q_id .== 111, :].text_answer[1]; #ans for Q2
+	sketch_data = filter(row -> row.uid == selected_student, s_df) #filtered sketch data frame containing the entries of the selected student
```

Comment:
Any time a comment wraps onto multiple lines, put it above the line of code instead of at the end of it